### DISCUSSION TITLE
include_ids - node/way/relation

### DISCUSSION THREAD

**Discussion by BenOnTrack** [systemed/tilemaker#613] [Q&A] [answered]
Is it fair to say that Tilemaker stores the id of the object, as long as `include_ids` is set to True, w/o having a way to differentiate the object itself ?

In other words, if I click on an object stored in a layer generated by Tilemaker, and get id = A, this could be either a node/way/relation and it's up to me to know what I'm looking for.

I can't remember which tile server (maptiler maybe?) does add a "prefix" to their ids, such as
- 0A -> node A
- 1A -> way A
- 2A -> relation A

Today I have a function:
```
function WriteOSMidAndType(obj, id_prefix)
	obj:Attribute("id", id_prefix .. obj:Id())
end
```
that is adding such a prefix - the only thing I don't like with this workaround is that any attribute written with Attribute function ends up in the "properties" of the object while the "include_ids" include the id at the root of the object. Any way to do something similar directly in the process.lua w/o the need to tweak the source code ?

PS: i have the feeling I've already asked a similar question but I can't put my hands on it. Sorry if i did.

**Comment by systemed** [comment, marked as answer]
Not at the moment. It wouldn't be crazily difficult to tweak tilemaker to do it though.

(Memo to self: the offset would need to be added before each call to `addGeometryToIndex` in `osm_lua_processing.cpp`, and it would be best controlled by making `include_ids` a trinary value of `true`/`false`/`"offset"`.)
